src/vs/code/node/cliVerboseLogger.ts

```diff
@@ -0,0 +1,42 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IDisposable } from 'vs/base/common/lifecycle';
+import { Promises } from 'vs/base/node/pfs';
+import { watchFile } from 'vs/base/node/watcher';
+
+/**
+ * This class provides logging for verbose commands on Mac
+ * as part of #102975
+ */
+export class CliVerboseLogger {
```

Comment:
I feel this code is now generic enough that we can simply move it into `src/vs/base/node/watcher.ts` simply as another method that is exposed. I would not call it `streamFile` because you are not just streaming but also watching. Maybe `watchFileStream`?